### PR TITLE
Add lease tests for edge

### PR DESCRIPTION
**Pull request by pdmars**
```
$ tempest run --exclude-list etc/exclude_list --concurrency 1 --regex "blazar_tempest_plugin.tests.api.leases.test_lease_devices"                              
{0} blazar_tempest_plugin.tests.api.leases.test_lease_devices.TestLeaseContainers.test_delete_lease_with_container [60.605183s] ... ok
{0} blazar_tempest_plugin.tests.api.leases.test_lease_devices.TestLeaseContainers.test_device_in_lease [46.446012s] ... ok
{0} blazar_tempest_plugin.tests.api.leases.test_lease_devices.TestLeaseContainers.test_device_not_in_multiple_leases [53.092664s] ... ok
{0} blazar_tempest_plugin.tests.api.leases.test_lease_devices.TestLeaseContainers.test_extend_lease_for_reserved_container [47.583610s] ... ok

======
Totals
======
Ran: 4 tests in 207.7352 sec.
 - Passed: 4
 - Skipped: 0
 - Expected Fail: 0
 - Unexpected Success: 0
 - Failed: 0
Sum of execute time for each test: 207.7275 sec.
```

### CODE REVIEW

File: src/blazar_tempest_plugin/tests/api/leases/test_lease_devices.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+        resp = self.leases_client.delete_lease(self.lease["id"])
+        self.assertEquals(200, resp.response.status)
+
+        try:
+            self.container_client.ensure_container_in_desired_state(
+                self.container.uuid, "Deleted"
+            )
+        except exceptions.NotFound:
+            pass
```
[the remark this answers is not in the document]
Comment (pdmars) [reply src/blazar_tempest_plugin/tests/api/leases/test_lease_devices.py:64]: Done, switched to adding those extra checks.

File: src/blazar_tempest_plugin/tests/api/leases/test_lease_devices.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+        try:
+            self.create_test_lease(
+                start_date=start,
+                end_date=end,
+                reservations=[device_reservation_request],
+            )
+            self.fail("Exception not raised when attempting to double-book the same device")
+        except exceptions.ServerFault as e:
+            self.assertIn("not enough resources available", str(e).lower())
```
[the remark this answers is not in the document]
Comment (pdmars) [reply src/blazar_tempest_plugin/tests/api/leases/test_lease_devices.py]: done

File: src/blazar_tempest_plugin/tests/api/leases/test_lease_devices.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        self.assertEquals(200, resp.response.status)
+
+        waiters.wait_for_lease_termination(self.leases_client, self.lease["id"])
+
+        try:
+            self.container_client.ensure_container_in_desired_state(
+                self.container.uuid, "Deleted"
+            )
+        except exceptions.NotFound:
+            pass
```
Comment (msherman64) [inline src/blazar_tempest_plugin/tests/api/leases/test_lease_devices.py:64]: just leaving a future TODO / nit here, we should probably handle this in a helper method somewhere, because try/except in the test classes is an antipattern.

No worries about it for now.

**Review by msherman64 — APPROVED** [review]
lgtm, thanks!

File: src/blazar_tempest_plugin/tests/api/leases/test_lease_devices.py (hunk 3 above)
Comment (pdmars) [reply src/blazar_tempest_plugin/tests/api/leases/test_lease_devices.py]: Not sure that I agree that using try in a test is always an antipattern :) In this case, I'd call it more of a "pragmatic workaround due to the eventually consistent nature of OpenStack APIs" :)

My bigger pet peeve of this code of mine is ignoring the exception, so I added a bit more to assert it more throughly.

Also happy to move this into a utility at some point as well.
